corehq/messaging/util.py

```diff
@@ -41,14 +49,26 @@ def rule_initiation_key_is_set(self):
     def rule_initiation_key_minutes_remaining(self):
         return (self.client.ttl(self.rule_initiation_key) // 60) or 1
 
-    def set_initial_progress(self):
+    def set_initial_progress(self, shard_count=0):
+        # shard_count is passed when tasks run pear each shard
         self.client.set(self.current_key, 0)
         self.client.set(self.total_key, 0)
+        if shard_count:
+            self.client.set(self.shard_count_key, shard_count)
```

Comment:
should we set the expiry on this key as well?

Reply:
Sure